src/Workspaces/Core/Portable/FindSymbols/FindReferences/BaseTypeFinder.cs

```diff
@@ -43,29 +39,39 @@ internal static partial class BaseTypeFinder
 
                     if (sourceMember.Symbol != null)
                     {
-                        // These are explicit interface implementations matching by name.
-                        if (type?.TypeKind == TypeKind.Interface)
-                        {
-                            if (symbol.ContainingType?.TypeKind == TypeKind.Class || symbol.ContainingType?.TypeKind == TypeKind.Struct)
-                            {
-                                var implementation = symbol.ContainingType.FindImplementations(sourceMember.Symbol, solution.Workspace);
-
-                                if (implementation != null &&
-                                    SymbolEquivalenceComparer.Instance.Equals(implementation.OriginalDefinition, symbol.OriginalDefinition))
-                                {
-                                    results.Add(sourceMember);
-                                }
-                            }
-                        }
-                        else if (SymbolFinder.IsOverride(solution, symbol, sourceMember.Symbol, cancellationToken))
+                        // Add to results overridden members only. Do not add hidden members.
+                        if (SymbolFinder.IsOverride(solution, symbol, sourceMember.Symbol, cancellationToken))
                         {
                             results.Add(sourceMember);
                         }
+
+                        // Add both overridden and hidden members.
+                        baseClassesMembers.Add(sourceMember.Symbol);
                     }
                 }
             }
 
-            return results.ToImmutableAndFree();
+            // This is called for all: class, struct or interface member.
+            results.AddRange(
+                await ConvertToSymbolAndProjectIdsAsync(
+                    symbol.ExplicitOrImplicitInterfaceImplementations(),
```

Comment:
i don't trust this call at all.  ExplicitOrImplicitInterfaceImplementations doesn't actually find *all* results because it doesn't know about this case:

```
class C
{
    public void Foo() { }
}

interface I { void Foo(); }

class D : C
{
}
```

in this case, C.Foo is an implmeentation of I.Foo.  

Note: this is an important case and we need to ensure it works in this codepath and we're not regressing anything.

Reply:
ok.  i'm a bit less concerned now that is ee that this is only called by the Find-Base feature.  I thought this was the general SymbolFinder logic which is used by a lot of features.

Reply:
I hope I specifically cover this with the iteration below over baseClassesMembers.

---
In reply to: [313679263](https://github.com/dotnet/roslyn/pull/37942#discussion_r313679263) [](ancestors = 313679263)

Reply:
@ivanbasov if you feel you have the right tests in place for *both* c# and vb for the case i mentioned (and your case as well), then i'm ok.  i'm not certain if those tests do exist though :)  but i leave it to you.  if you've checked and are good, then 'm good :)

Reply:
In the C# scenario you mentioned, we definitely find I.Foo for C.Foo. However, there is no D.Foo. Therefore, we cannot find a base for it.
If D would have one, there are 
TestWithVirtualMethodHiddenWithInterfaceOnDerivedClass
and TestWithVirtualMethodHiddenAndInterfaceImplementedOnDerivedType

---
In reply to: [314057545](https://github.com/dotnet/roslyn/pull/37942#discussion_r314057545) [](ancestors = 314057545)

Reply:
>In the C# scenario you mentioned, we definitely find I.Foo for C.Foo.

Great.  As long as we're making the implicit connection that C.Foo impelments I.Foo through `D: C`  (and we test for this), the i'm happy :)